lib/determinator/cache/fetch_wrapper.rb

```diff
@@ -4,19 +4,24 @@ class FetchWrapper
       # @param *caches [ActiveSupport::Cache] If a list then the head of the the
       # list should will be checked before the  tail. If the head is empty but
       # the tail is not then the head will be filled with the value of the tail.
-      def initialize(*caches)
+      def initialize(*caches, cache_missing: true)
+        @cache_missing = cache_missing
         @caches = caches
       end
 
       # Call walks through each cache, returning a value if the item exists in
       # any cache, otherwise popularing each cache with the value of yield.
       def call(feature_name)
         value = read_and_upfill(feature_name)
-        # nil is an acceptable value in the case of a missing feature definition
-        return nil if value.nil?
-        return value if value != false
+
+        # if the value is missing and we cache it, return the missing response
+        return value if value.is_a?(MissingResponse) && @cache_missing
+
+        #otherwise only return the non nil/notice_missing_feature value
+        return value if !value.nil? && !(value.is_a?(MissingResponse) && !@cache_missing)
```

Comment:
Since this now can never be nil (thanks), I wonder if should capture the `ErrorResponse` instead. Also the not of several "not"s makes this a bit complicated to read

Reply:
Yeah, I'll follow this up with a second PR to add an error reporter for these. For now they'll be reported as missing.